here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/AbstractTask.java

```diff
@@ -54,16 +51,23 @@ public abstract class AbstractTask<RESULT, SELF extends AbstractTask<RESULT, SEL
     implements INakshaBound, UncaughtExceptionHandler {
 
   private static final Logger log = LoggerFactory.getLogger(AbstractTask.class);
-  static int thresholdPerProcessor;
-  static int percentageThresholdPerPrinciple;
+  private static int maxParallelRequestsPerCPU = 50;
+  private static int maxPctParallelRequestsPerPrincipal = 100;
 
   public static void initConcurrencyLimits(int cpuThreshHold, int perPrincipalThreshold) {
-    thresholdPerProcessor = cpuThreshHold;
-    percentageThresholdPerPrinciple = perPrincipalThreshold;
+    maxParallelRequestsPerCPU = cpuThreshHold;
+    maxPctParallelRequestsPerPrincipal = perPrincipalThreshold;
   }
 
-  private static final Map<String, AtomicInteger> authorUsageMap = new ConcurrentHashMap<>();
+  /**
+   * The soft-limit of tasks to run concurrently. This limit does normally not apply to child tasks.
+   */
+  public static final AtomicLong limit =
+      new AtomicLong((long) Runtime.getRuntime().availableProcessors() * maxParallelRequestsPerCPU);
```

Comment:
Why should this be an atomic long, does it change at runtime? Do do not find any changing code. If not, make it a `final long` (very likely we do not want it to be changed at runtime or?)

Reply:
Fixed.